src/librustc_typeck/check/mod.rs

```diff
@@ -1311,9 +1311,40 @@ fn check_union(tcx: TyCtxt<'_>, id: hir::HirId, span: Span) {
     def.destructor(tcx); // force the destructor to be evaluated
     check_representable(tcx, span, def_id);
     check_transparent(tcx, span, def_id);
+    check_union_fields(tcx, span, def_id);
     check_packed(tcx, span, def_id);
 }
 
+/// When the `#![feature(untagged_unions)]` gate is active,
+/// check that the fields of the `union` does not contain fields that need dropping.
+fn check_union_fields(tcx: TyCtxt<'_>, _: Span, item_def_id: DefId) -> bool {
+    // Without the feature we check Copy types only later
```

Comment:
Where is "later"? A reference would be nice.

Reply:
Is there a good reason these two checks are done in totally different places? Doing them in the same place would be better IMO.

Reply:
I have no idea. This code comes from https://github.com/rust-lang/rust/pull/56440.

Reply:
Could you find out?

Reply:
I adjusted the comment

The actual check is happening in https://github.com/rust-lang/rust/blob/f7434aef26bc413469fc0dfeadecabf75e18bd38/src/librustc/middle/stability.rs#L834